src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -2235,8 +2275,18 @@ int xclLoadXclBin(xclDeviceHandle handle, const xclBin *buffer)
 #endif
 
     if (!ret) {
+      char *pMemInfo = NULL;
       auto core_device = xrt_core::get_userpf_device(drv);
-      core_device->register_axlf(buffer);
+      
+      /* Get memory topology info from driver through sysfs */
+      drv->xclGetMemInfo(&pMemInfo);
+      /* Populate the retrive info into device class */
+      core_device->register_axlf(buffer, pMemInfo);
```

Comment:
What is pMemInfo?  All the sections are part of the xclbin or should be accessed via query requests, if group_topology/connectivity are not in the xclbin or not in xclbin, then register_axlf should simply copy mem_topology and connectivity.  I don't think register_axlf signature should change.

Reply:
Removed this function.